src/app/modules/main/profile_section/wallet/networks/combined_model.nim

```diff
@@ -0,0 +1,74 @@
+import NimQml, Tables, strutils, strformat
+
+import ./combined_item
+
+type
+  ModelRole* {.pure.} = enum
+    Prod = UserRole + 1,
+    Test
+
+QtObject:
+  type
+    CombinedModel* = ref object of QAbstractListModel
+      items: seq[CombinedItem]
+
+  proc delete(self: CombinedModel) =
+    self.items = @[]
+    self.QAbstractListModel.delete
+
+  proc setup(self: CombinedModel) =
+    self.QAbstractListModel.setup
+
+  proc newCombinedModel*(): CombinedModel =
+    new(result, delete)
+    result.setup
+
+  proc `$`*(self: CombinedModel): string =
+    for i in 0 ..< self.items.len:
+      result &= fmt"""[{i}]:({$self.items[i]})"""
+
+  proc countChanged(self: CombinedModel) {.signal.}
+
+  proc getCount(self: CombinedModel): int {.slot.} =
+    self.items.len
+
+  QtProperty[int] count:
+    read = getCount
+    notify = countChanged
+
+  method rowCount*(self: CombinedModel, index: QModelIndex = nil): int =
+    return self.items.len
+
+  method roleNames(self: CombinedModel): Table[int, string] =
+    {
+      ModelRole.Prod.int:"prod",
+      ModelRole.Test.int:"test",
```

Comment:
I'd find it better (and more logical imho) to keep the old model with all its roles and just extend it with this `prod/test` boolean, wdyt?

Reply:
@Khushboo-dev-cpp as I see the only fields which differ between live and test network are:
`rpcURL`, `fallbackURL`, `blockExplorerURL`

So as @caybro suggested you could just update/extend old item to have these fields:
`mainRpcURL`
`mainFallbackURL`
`mainBlockExplorerURL`
`testRpcURL`
`testFallbackURL`
`testBlockExplorerURL`

Reply:
@saledjenic the icons are going to be different as well once Bens designs are ready and maybe something else in the future. So I'd prefer not adding two properties for both of these as it can keep growing? wdyt?

Reply:
It's up to you, I personally do not like duplications. What I see as the best case here is the following, separate item for networks specific details:
```
type
  NetworkDetailsItem* = object
     rpcURL: string
     fallbackURL: string
     blockExplorerURL: string
```
and separate item for network item:
```
type
  NetworkItem* = object
    chainId: int
    layer: int
    chainName: string
    iconUrl: string
    shortName: string
    chainColor: string
    mainNetwork: NetworkDetailsItem
    testNetwork: NetworkDetailsItem
```
later on if Ben adds different items for test and main network we will just move `iconUrl` from `NetworkItem` to `NetworkDetailsItem`, easily.